activestorage/lib/active_storage/attached/many.rb

```diff
@@ -28,12 +28,14 @@ def blobs
     #   document.images.attach(io: File.open("/path/to/racecar.jpg"), filename: "racecar.jpg", content_type: "image/jpeg")
     #   document.images.attach([ first_blob, second_blob ])
     def attach(*attachables)
-      if record.persisted? && !record.changed?
-        record.public_send("#{name}=", blobs + attachables.flatten)
-        record.save
-      else
-        record.public_send("#{name}=", (change&.attachables || blobs) + attachables.flatten)
+      if Array(attachables).any?
+        record.attachment_changes[name] = ActiveStorage::Attached::Changes::CreateMany.new(name,
+                                                                                           record,
+                                                                                           blobs + attachables.flatten,
+                                                                                           pending_upload: change&.try(:pending_upload))
```

Comment:
I think this logic of passing the pending uploads to the change object would be in the attribute setter definition in `ActiveStorage::Attached::Model::Model#has_many_attached`:

https://github.com/rails/rails/blob/4f191b975431dbf8071d73a48505e27b19c283e6/activestorage/lib/active_storage/attached/model.rb#L147

That way, we keep the concern of instantiating `ActiveStorage::Attached::Changes::CreateMany` in a single place. Also we could lean on the same `#{name}=` method here.